### PR TITLE
feat(log): introduce emperror for errors with details

### LINKED ISSUE #797: Evaluate switching to emperror
##### What can be improved

One error interface for wrapping, multi error and detailed errors

##### Why

* multi error
* detailed error for logging
* wrapping with stack trace

https://github.com/emperror/errors

### PR DESCRIPTION
**Pull request by aslakknutsen**
* include details in error where possible
* only include stacktrace at the lowest level

Fixes #797

### CODE REVIEW

File: pkg/log/emperror.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+			for n := i + 2; n < len(kv); n += 2 {
+				if v2, ok := kv[n].(string); ok {
+					if v == v2 {
+						kv = append(kv[:n], kv[n+2:]...)
+						n -= 2
+					}
+				}
+			}
+		}
+	}
```
Comment (bartoszmajsak) [inline pkg/log/emperror.go:35]: image

File: pkg/log/emperror_test.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
+package log_test
```
Comment (bartoszmajsak) [inline pkg/log/emperror_test.go:1]: How are we testing things here? :)

**Review by bartoszmajsak — APPROVED** [review]
Just two points to bring to your attention. And a gif :)

File: pkg/log/emperror_test.go (hunk 2 above)
Comment (aslakknutsen) [reply pkg/log/emperror_test.go:1]: Yeah, wanted to bring that up. We're not testing anything. But it's a 'structure' for manual verification? Not sureif we should just dump it or what to do with it

Comment (bartoszmajsak) [reply pkg/log/emperror_test.go:1]: Ok lets keep it like that but maybe follow-up with https://github.com/maistra/istio-workspace/issues/833 should be considered as good time of evaluating it. WDYT?

Comment (aslakknutsen) [reply pkg/log/emperror_test.go:1]: That sounds good

**Pull request merged**